tests/Unit/RemoveFromSearchTest.php

```diff
@@ -54,20 +54,20 @@ public function test_models_are_deserialized_without_the_database()
         $this->assertEquals(1234, $job->models->first()->getScoutKey());
     }
 
-    public function test_models_are_deserialized_without_the_database_using_custom_scout_key()
+    public function test_models_are_deserialized_without_the_database_using_custom_scout_key_values()
     {
         $job = new RemoveFromSearch(Collection::make([
-            $model = new SearchableModelWithCustomKey(['other_id' => 1234]),
+            $model = new MeiliSearchCustomKeySearchableModel(['id' => 1234]),
         ]));
 
         $job = unserialize(serialize($job));
 
         $this->assertInstanceOf(Collection::class, $job->models);
         $this->assertCount(1, $job->models);
-        $this->assertInstanceOf(SearchableModelWithCustomKey::class, $job->models->first());
+        $this->assertInstanceOf(MeiliSearchCustomKeySearchableModel::class, $job->models->first());
         $this->assertTrue($model->is($job->models->first()));
-        $this->assertEquals(1234, $job->models->first()->getScoutKey());
-        $this->assertEquals('searchable_model_with_custom_keys.other_id', $job->models->first()->getScoutKeyName());
+        $this->assertEquals('my-meilisearch-key.1234', $job->models->first()->getScoutKey());
+        $this->assertEquals('meili_search_custom_key_searchable_models.id', $job->models->first()->getScoutKeyName());
```

Comment:
This test will pass because the `MeiliSearchCustomKeySearchableModel` fixture is being instantiated with primary key that is being used in the `getScoutKey()` method:

```php
public function getScoutKey()
{
    return 'my-meilisearch-key.'.$this->getKey();
}
```

https://github.com/DarronEngelbrechtEdge/scout/blob/246bf96e1042f1b1adc586f69f6260e389a1dccb/tests/Unit/MeiliSearchEngineTest.php#L475-L478

If another model attribute is used to construct the scout key, then this will fail, as it cannot be retrieved from the faux model that is created after the job is unserialized. All other model properties will not exist.